src/plugins/ssh/index.ts

```diff
@@ -273,7 +287,7 @@ const addCommonArgs = (args: CommandArgs, proxyCommand: string[]) => {
       opt === "-o" && sshOptions[idx + 1]?.startsWith("ProxyCommand")
   );
 
-  if (!proxyCommandExists) {
+  if (!proxyCommandExists && proxyCommand.length > 0) {
```

Comment:
nit: could make the new `proxyCommand.length > 0` condition part of `proxyCommandExists`

Reply:
These are actually two different checks so I don't want to combine them. `proxyCommandExists` is checking whether there is a `ProxyCommand` already specified by the user as an argument to `p0 ssh` (e.g., `p0 ssh ... -- -o 'ProxyCommand /some/command`) so we don't overwrite it. I'm not sure what the use case for this behavior is, but I figure it's out of scope to touch this. `proxyCommand.length > 0` is checking whether the `SshProvider` in use specifies a `ProxyCommand`, of which the Azure provider is the only one that doesn't. Without the check, we will try to append an empty `ProxyCommand` to the SSH arguments (e.g., `-o 'ProxyCommand'`) which is an error.

What I will do though is do some renaming here to make this more clear, but let me know if you strongly think they should be combined.

Reply:
I see, I thought `proxyCommandExists` is whether the provider has it. The new names are clear! 🙏